code/modules/atmospherics/machinery/components/fusion/hfr_main_processes.dm

```diff
@@ -516,16 +504,17 @@
 			else
 				filtering = FALSE
 		if(filtering && moderator_internal.gases[filter_type])
-			var/datum/gas_mixture/removed = moderator_internal.remove_specific(filter_type, 20 * delta_time)
+			var/datum/gas_mixture/removed = moderator_internal.remove_specific(filter_type, 100 * delta_time)
 			if(removed)
 				linked_output.airs[1].merge(removed)
 
 		var/datum/gas_mixture/internal_remove
-		if(internal_fusion.gases[/datum/gas/helium][MOLES] > 0)
-			internal_remove = internal_fusion.remove_specific(/datum/gas/helium, internal_fusion.gases[/datum/gas/helium][MOLES] * (1 - (1 - 0.5) ** delta_time))
-			linked_output.airs[1].merge(internal_remove)
+		for(var/gas_id in selected_fuel.primary_products)
+			if(internal_fusion.gases[gas_id][MOLES] > 0)
+				internal_remove = internal_fusion.remove_specific(gas_id, internal_fusion.gases[gas_id][MOLES] * (1 - (1 - 0.25) ** delta_time))
+				linked_output.airs[1].merge(internal_remove)
 		if(internal_fusion.gases[/datum/gas/antinoblium][MOLES] > 0)
-			internal_remove = internal_fusion.remove_specific(/datum/gas/antinoblium, internal_fusion.gases[/datum/gas/antinoblium][MOLES] * (1 - (1 - 0.05) ** delta_time))
+			internal_remove = internal_fusion.remove_specific(/datum/gas/antinoblium, internal_fusion.gases[/datum/gas/antinoblium][MOLES] * (1 - (1 - 0.25) ** delta_time))
```

Comment:
?

Reply:
had to increase the antinoblium production a bit, since it was too low to make anything meaningful